modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/implementation/clientExtensionFilter.tsx

```diff
@@ -0,0 +1,188 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ClientExtension} from 'frontend-js-components-web';
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+
+import type {FDSFilter} from '@liferay/js-api/data-set';
+
+import type {
+	FilterImplementation,
+	FilterImplementationArgs,
+	SetFilterArgs,
+} from '../Filter';
+
+export interface ClientExtensionFilterImplementationArgs
+	extends FilterImplementationArgs<unknown> {
+	cxFilterImplementation?: FDSFilter<unknown>;
+	cxFilterURL: string;
+}
+
+function getSelectedItemsLabel({
+	cxFilterImplementation,
+	cxFilterURL,
+	selectedData,
+}: ClientExtensionFilterImplementationArgs) {
+	if (!cxFilterImplementation) {
+		return '...';
+	}
+
+	if (!cxFilterImplementation.descriptionBuilder) {
+		console.warn(
+			'The filter client extension',
+			cxFilterURL,
+			'is not exporting a descriptionBuilder() function,',
+			'thus the filter description will not show meaningful information.',
+			'The client extension should be reworked.'
+		);
```

Comment:
A lot of console use makes me a bit nervous. I'm not categorically against it, but we tend to use it very sparingly. If we covered all cases where something goes unexpected, dev tools would likely be flooded with warnings.

Reply:
One thing I proposed to @bryceosterhaus very long ago was creating a loggin API for the frontend side. It would be something simple, similar to LOG4J and it would allow you to tweak the levels from system settings and/or a JS API, maybe. Something like that...

We definitely are going to need something like that, because the only way we can inform users of FE CX errors is using the console, and we want to inform them since, otherwise, there won't be any easy way to diagnose this things and it will flood us with support requests and escalations :see_no_evil: 

In any case, I would go with the code as it is now, because this should be a strange condition. I mean, if a CX is broken is most probably because you are developing it, so you definitely want to see the error in the console. If it fails in production, well, flooding the console will be the smallest of your problems...

One thing we may do in this case is emitting each message just once (a logging API will come in handy for implementing this globally). IINM, you can get several error messages from the `getSelectedItemsLabel` function, for example, because it is invoked several times...

Reply:
>the only way we can inform users of FE CX errors is using the console

Is the user you are referring to us (internal devs) or customers?

If it's us, it's not a common practice to add logs, but if you think it's valuable, I am not against it. It should only be wrapped in dev environment check, like [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js#L108-L113).

If it's customers, then logging is a product question, something we should treat as a new feature to refine. It's a bit of a can of worms, as I would expect customers to (1) require localization and (2) require consistent and high quality logs in all CX implementations.

Reply:
Yes, it's true that it should be a properly implemented feature but until that arrives, it's better to have a not perfect log than nothing at all. I've spent countless hours in support tasks simply because the product didn't signal errors and it's a place I don't want to revisit :sweat_smile: .

In fact, users complain every once in a while about the poor logging of errors DXP implements server side in some of its areas :-(

Reply:
>  (1) require localization 

This shouldn't be needed since we are talking about technical errors. The server side doesn't localize errors, for instance, it just uses English, as most products do.

> (2) require consistent and high quality logs in all CX implementations.

This is true, but we have to begin somewhere and this CX may be a good place to start, since it's the first that needs to be lenient when the CX fails to load.

What we can do is (once this is settled) to create a simple logging API that lets us tweak logging levels and/or categories to make it easier to understand what's going on. Something similar to LOG4J for the browser side should be more than enough IMO...